js/sketch.js

```diff
@@ -0,0 +1,86 @@
+class Helper {
+    generateRandomNumber(min, max) {
+        return Math.floor(Math.random() * (max - min + 1)) + min;
+    }
+
+    generateRandomColor() {
+        return [
+            this.generateRandomNumber(0, 255),
+            this.generateRandomNumber(0, 255),
+            this.generateRandomNumber(0, 255),
+        ];
+    }
+}
+
+class Circle {
+    x;
+    y;
+    size;
+    stepsX;
+    stepsY;
+
+    constructor(x, y, size, stepsX, stepsY) {
+        this.x = x;
+        this.y = y;
+        this.size = size;
+        this.stepsX = stepsX;
+        this.stepsY = stepsY;
+        this.color = new Helper().generateRandomColor();
+    }
+
+    createCircle() {
+        circle(this.x, this.y, this.size);
+    }
+
+    fillColor() {
+        fill(
+            `rgba(${this.color[0]}, ${this.color[1]}, ${this.color[2]}, 0.5)`
+        );
+    }
+
+    moveCircle() {
+        this.x += this.stepsX;
+        this.y += this.stepsY;
+
+        if (this.x > width || this.x < 0) {
+            this.stepsX *= -1;
+            this.color = new Helper().generateRandomColor();
+        }
+
+        if (this.y > height || this.y < 0) {
+            this.stepsY *= -1;
+            this.color = new Helper().generateRandomColor();
+        }
+    }
+}
+
+let circles = [];
+const numberOfCircles = 20;
+const helper = new Helper();
+
+function preload() {
+    background = loadImage("../images/background-image.jpg");
+}
+
+function setup() {
+    createCanvas(windowWidth, windowHeight);
+
+    for (let i = 0; i < numberOfCircles; i++) {
```

Comment:
maybe a cleaner way to do this is to define a empty array and foreach going through it and mapping the content of it 🤔 
```suggestion
    let circles = Array(20).fill(null);
    circles.map(() => new Circle(
    helper.generateRandomNumber(0, width), ...
```